### PR TITLE
[SYCL][NFC] Workaround for bug in GCC older than 7

### PR DESCRIPTION
**Pull request by vladimirlaz**
https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480

Signed-off-by: Vladimir Lazarev <vladimir.lazarev@intel.com>

### CODE REVIEW

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 1]
```diff
@@ -169,25 +169,28 @@ void printArgs(Arg0 arg0, Args... args) {
 
 namespace RT = cl::sycl::detail::pi;
 
+// Workaround for build with GCC 5.5.0
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480)
+namespace pi {
 // Want all the needed casts be explicit, do not define conversion
 // operators.
-template <class To, class From> To inline pi::cast(From value) {
+template <class To, class From> To inline cast(From value) {
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.hpp]:
```suggestion
template <class To, class From> inline To cast(From value) {
```

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 2]
```diff
@@ -169,25 +169,28 @@ void printArgs(Arg0 arg0, Args... args) {
 
 namespace RT = cl::sycl::detail::pi;
 
+// Workaround for build with GCC 5.5.0
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.hpp]:
```suggestion
// Workaround for build with GCC 5.x
```

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 3]
```diff
@@ -169,25 +169,28 @@ void printArgs(Arg0 arg0, Args... args) {
...
+// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480)
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.hpp]:
```suggestion
// https://gcc.gnu.org/bugzilla/show_bug.cgi?id=56480
```
Could you add a short description of the problem and solution in the comments, so reader will not have to go to the link in order to understand what does it mean?

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 4]
```diff
@@ -169,25 +169,28 @@ void printArgs(Arg0 arg0, Args... args) {
...
-template <class To, class From> To inline pi::cast(From value) {
+template <class To, class From> To inline cast(From value) {
   // TODO: see if more sanity checks are possible.
   RT::assertion((sizeof(From) == sizeof(To)), "assert: cast failed size check");
   return (To)(value);
 }
 
 // These conversions should use PI interop API.
-template <> pi::PiProgram inline pi::cast(cl_program interop) {
+template <> pi::PiProgram inline cast(cl_program interop) {
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.hpp]:
```suggestion
template <> inline pi::PiProgram cast(cl_program interop) {
```

File: sycl/include/CL/sycl/detail/pi.hpp [hunk 5]
```diff
@@ -169,25 +169,28 @@ void printArgs(Arg0 arg0, Args... args) {
...
   RT::assertion(false, "pi::cast -> use piextProgramConvert");
   return {};
 }
 
-template <> pi::PiDevice inline pi::cast(cl_device_id interop) {
+template <> pi::PiDevice inline cast(cl_device_id interop) {
```
Comment (bader) [inline sycl/include/CL/sycl/detail/pi.hpp]:
```suggestion
template <> inline pi::PiDevice cast(cl_device_id interop) {
```

**Comment by bader** [comment]
@vladimirlaz, don't see any changes. Please, update the patch.

**Comment by vladimirlaz** [comment]
> @vladimirlaz, don't see any changes. Please, update the patch.
@bader, sorry source part of the change were not uploaded. No it has been done.

**Pull request merged**